### PR TITLE
Converts server-routerlicious.yml pipeline to 1ES template

### PR DESCRIPTION
**Pull request by seanimam**
## Description

This PR converts server-routerlicious.yml to the new 1ES template required by Microsoft with the minimal required set of template changes required. This new template enables security and auditing checks by Microsoft. Read more on how the logic for converting to a 1ES pipeline works on [the official 1ES wiki](https://eng.ms/docs/cloud-ai-platform/devdiv/one-engineering-system-1es/1es-docs/1es-pipeline-templates/onboarding/overview)

A temporary 1ES folder was created for converted pipeline templates. This enables other pipelines to remain unchanged and use the original templates while new PR's are opened for individually converted other pipelines to 1ES as required by Microsoft. **You can easily view what the template changes are by looking at the diff in [the following PR](https://github.com/microsoft/FluidFramework/pull/19384) which is a fork of this branch and simply moves the template changes from the 1ES folder to the original so there is a clear diff present on github**

- Once all pipelines are moved over, we can start moving the files out of the 1ES folder and eventually remove the folder since it will no longer have any files (This is after we convert the rest of the required pipelines). 

- This PR also adds a new file CredScanSuppressions.json which disables automated credential scanning security checks on certain files that produce a false positive because they have mock credentials.

## Reviewer Guidance

- Confirm the newly converted pipelines run successfully. Look at the completed steps and compare with a recent successful run on main to ensure the expected steps run (plus the new SDL and 1ES security checks)
- Make sure all the relevant templates and pipelines for server-routerlicious.yml have been converted
- Make sure there are no unnecessary changes or 1ES template and/or pipeline conversions
- Make sure the yml is valid -- a successful pipeline run should suffice for confirming this

### CODE REVIEW

File: tools/pipelines/server-routerlicious.yml [hunk 1]
```diff
@@ -96,12 +96,28 @@ pr:
...
     - server/routerlicious/kubernetes/routerlicious
 
+variables:
+  - template: /tools/pipelines/templates/1ES/include-vars.yml@self
+    parameters:
+      releaseImage: true
+      publishOverride: ${{ parameters.publishOverride }}
+      releaseBuildOverride: ${{ parameters.releaseBuildOverride }}
+      buildNumberInPatch: false
+  # Note: we tried to centralize the use of the 'container-registry-info' variable group (i.e. put it inside the shared
```
Comment (Josmithr) [inline tools/pipelines/server-routerlicious.yml:106]: Thanks for the comment! Have we reported this deficiency? It isn't immediately clear to me if this is something we'll be able to clean up in the future.

**Review by Josmithr — APPROVED** [review]

**Review by alexvy86 — CHANGES_REQUESTED** [review]
Just to make sure we don't merge this before addressing https://github.com/microsoft/FluidFramework/pull/19384#discussion_r1471705880

**Review by alexvy86 — APPROVED** [review]

**Comment by alexvy86** [comment]
Found one more thing we need to bring with this (could have done it with any of them really, because the relevant file is already getting analyzed as part of the repo): [the CredScanSuppressions.json file](https://github.com/microsoft/FluidFramework/pull/18397/files#diff-9b9d23f502279183135dd95b110ea9a0b18a3ef1b1e4bfab255a5c7936b10802). Due to the missing suppression file, the pipelines are reporting warnings [like this one](https://dev.azure.com/fluidframework/public/_build/results?buildId=233104&view=logs&j=bc38e8b8-e027-53cb-48e7-2adbd1070eca&t=d1aba2c3-5b21-511d-f336-affeabd629f9&l=53).

**Review by alexvy86 — APPROVED** [review]
🚀

**Pull request merged**